res/controllers/Reloop-Ready-scripts.js

```diff
@@ -475,485 +478,467 @@ ReloopReady.AutoLoopPadMode = function(index) {
+                    this.inSetValue(loopMoveBeats);
+                });
+            },
+            shift: function() {
+                this.input = ReloopReady.makeButtonDownInputHandler(function() {
+                    this.inSetValue(engine.getValue(this.group, "loop_size"));
+                });
+            },
+            trigger: ReloopReady.makeIsPressedTrigger(),
+        });
+
+        this.pads.forEach((pad, i) => pad.midi = [0x94 + deckIdx, 0x14 + i]);
 
-    // var baseOffset = index * 8;
+        this.parameterLeft = new components.Button({
+            midi: [0x94 + deckIdx, 0x28],
+            shiftOffset: 0x2,
+            input: ReloopReady.makeButtonDownInputHandler(function() {
+                loopMoveBeats *= 0.5;
+            }),
+        });
+        this.parameterRight = new components.Button({
+            midi: [0x94 + deckIdx, 0x29],
+            shiftOffset: 0x2,
+            input: ReloopReady.makeButtonDownInputHandler(function() {
+                loopMoveBeats *= 2;
+            }),
+        });
 
-    this.pads = this.pads.map((_, i) =>
-        new components.SamplerButton({
-            midi: [0x94 + index, 0x14 + i],
-            // number: baseOffset + i + 1,
+        this.reconnectComponents(c => {
+            if (c.group === undefined) {
+                c.group = group;
+            }
+        });
+    }
+};
+ReloopReady.SamplerPadMode = class extends ReloopReady.PadMode {
+    constructor(deckIdx) {
+        super(deckIdx);
+        this.pads = this.pads.map((control, i) => new components.SamplerButton({
+            midi: [0x94 + deckIdx, control],
             number: i + 1,
             shiftOffset: 0x8,
             off: 0x00,
             outConnect: false,
-        })
-    );
+        }));
+    }
 };
-ReloopReady.SamplerPadMode.prototype = Object.create(ReloopReady.PadMode.prototype);
-
-
-ReloopReady.LoopRollPadMode = function(index) {
-    ReloopReady.PadMode.call(this);
-
-    this.currentLoopSizeExp = -2;
-    const theContainer = this;
-
-    const clampLoopSizeExp = loopSizeExp => _.clamp(loopSizeExp, -5, -2);
-
-    this.setLoopSizes = loopSizeExp => {
-        _(_.range(loopSizeExp, loopSizeExp + 8))
-            .map(function(exp) { return Math.pow(2, exp); })
-            .zip(this.pads)
-            .forEach(([size, pad]) => {
-                pad.inKey = `beatlooproll_${size}_activate`;
-                pad.outKey = `beatloop_${size}_enabled`;
-            });
-    };
-
-    for (let i = 0; i < this.pads.length; i++) {
-        this.pads[i] = new components.Button({
-            midi: [0x94 + index, 0x14 + i],
-            group: `[Channel${index + 1}]`,
-            shiftOffset: 0x8,
-            off: ReloopReady.padColorPalette.Green.dim,
-            on: ReloopReady.padColorPalette.Green.lit,
-            outConnect: false,
-        });
+ReloopReady.LoopRollPadMode = class extends ReloopReady.AbstractLoopPadMode {
```

Comment:
Regression? The funky *Loop Roll* mode doesn't work since this commit. It worked before.

Reply:
not sure. I'll look into it once I have access to the hardware again.

Reply:
I have the hardware again and I can confirm this. Looking into it.